internal/services/network/application_gateway_resource.go

```diff
@@ -908,14 +1117,16 @@ func resourceApplicationGateway() *pluginsdk.Resource {
 							Type:     pluginsdk.TypeString,
 							Required: true,
 							ValidateFunc: validation.StringInSlice([]string{
-								string(network.ProtocolHTTP),
-								string(network.ProtocolHTTPS),
+								string(network.ApplicationGatewayProtocolHTTP),
+								string(network.ApplicationGatewayProtocolHTTPS),
+								string(network.ApplicationGatewayProtocolTCP),
+								string(network.ApplicationGatewayProtocolTLS),
 							}, false),
 						},
 
 						"path": {
 							Type:     pluginsdk.TypeString,
-							Required: true,
+							Optional: true,
```

Comment:
Is there a specific reason to change the `path` to be optional? If yes, then can be add a new test case with `path` not set?

Reply:
Path cannot be set for TLS/TCP.